src/main/kotlin/org/rust/lang/core/completion/RsImplTraitMemberCompletionProvider.kt

```diff
@@ -0,0 +1,127 @@
+            parentItems.removeIf { it.javaClass == item.javaClass && it.name == item.name }
+        }
+
+        for (item in parentItems) {
+            val lookup = getCompletion(item, trait, implBlock) ?: continue
+            result.addElement(
+                lookup.withPriority(KEYWORD_PRIORITY + 1)
+            )
+        }
+    }
+}
+
+private fun getCompletion(
+    target: RsAbstractable,
+    trait: BoundElement<RsTraitItem>,
+    impl: RsImplItem
+): LookupElementBuilder? {
+    val substitution = trait.subst
+    return when (target) {
+        is RsConstant -> completeConstant(target, substitution)
+        is RsTypeAlias -> completeType(target)
+        is RsFunction -> completeFunction(target, trait, impl)
+        else -> error("unreachable")
+    }
+}
+
+private fun completeConstant(target: RsConstant, substitution: Substitution): LookupElementBuilder? {
+    val text = buildString {
+        append("const ")
+        append(target.name)
+        val type = target.typeReference?.type?.substitute(substitution) ?: return null
+        append(": ")
+        append(type.renderInsertionSafe())
+        val builder = RsDefaultValueBuilder(target.knownItems, target.containingMod, RsPsiFactory(target.project))
+        append(" = ")
+        append(builder.buildFor(type, mapOf()).text)
+        append(';')
+    }
+    return LookupElementBuilder.create(text)
+        .withIcon(target.getIcon(0))
+        .withInsertHandler { context, _ ->
+            val element = context.getElementOfType<RsConstant>() ?: return@withInsertHandler
+            val expr = element.expr ?: return@withInsertHandler
+            selectElement(expr, context.editor)
+        }
+}
+
+private fun completeType(target: RsTypeAlias): LookupElementBuilder {
+    val text = buildString {
+        append("type ")
+        append(target.name)
+        append(" = ();")
+    }
+    return LookupElementBuilder.create(text)
+        .withIcon(target.getIcon(0))
+        .withInsertHandler { context, _ ->
+            val element = context.getElementOfType<RsTypeAlias>() ?: return@withInsertHandler
+            val typeReference = element.typeReference ?: return@withInsertHandler
+            selectElement(typeReference, context.editor)
+        }
+}
+
+private fun completeFunction(
+    target: RsFunction,
+    trait: BoundElement<RsTraitItem>,
+    impl: RsImplItem
+): LookupElementBuilder {
+    val substitution = trait.subst
+    val signature = target.getSignatureText(substitution, trait, impl) ?: ""
+    val text = "$signature{\n        unimplemented!()\n    }"
+    return LookupElementBuilder
+        .create(text)
+        .withIcon(target.getIcon(0))
+        .withInsertHandler { context, _ ->
+            val element = context.getElementOfType<RsFunction>() ?: return@withInsertHandler
+            val body = element.block?.expr ?: return@withInsertHandler
+            selectElement(body, context.editor)
+        }
```

Comment:
What's about auto import here?
For example, try to complete `take` method of `std::io::Read`
Currently, `Take` is not imported
![image](https://user-images.githubusercontent.com/2539310/122601256-ad2d9d00-d079-11eb-9d52-067f142f7c2b.png)
But don't insist to do it in this PR

Reply:
Thanks, I added it in https://github.com/intellij-rust/intellij-rust/pull/6342/commits/3d0807248d7e77a730f2fcf8e5c7393e3d32f2f8.